libs/pageserver_api/src/controller_api.rs

```diff
@@ -74,6 +74,17 @@ pub struct TenantPolicyRequest {
     pub scheduling: Option<ShardSchedulingPolicy>,
 }
 
+#[derive(Serialize, Deserialize)]
+pub struct ShardsPreferredAzsRequest {
+    #[serde(flatten)]
+    pub preferred_az_ids: HashMap<TenantShardId, String>,
```

Comment:
Is there a use case for having different preferred AZs for different shards?  I was thinking the AZ preference is always aimed at saying "this is the best AZ to run a compute", hence a per-tenant thing

Internally we store it per-shard as a convenience to avoid needing separate per-tenant and per-shard tables, but in the API we could just expose it as a per-tenant thing.

Reply:
> Is there a use case for having different preferred AZs for different shards?
Not right now, but I didn't want to rule it out. It's a low level endpoint, I think it's fine for it to operate on tenant shards. Storcon can wrap it and operate on tenants.